library/bdAd/Listener.php

```diff
@@ -77,4 +79,25 @@ public static function helperEngine()
 
         return call_user_func_array($method, $args);
     }
-}
\ No newline at end of file
+
+    public static function front_controller_pre_dispatch(XenForo_FrontController $fc, XenForo_RouteMatch &$routeMatch)
+    {
+        $routeMatch->getMajorSection();
+        $majorSection = $routeMatch->getMajorSection();
+
+        $checkedValues = bdAd_Option::get('customizeDisplayAd');
+
+        foreach ($checkedValues as $key => $value) {
+            if ($majorSection == $key) {
+                self::$_noAd = false;
+            }
+        }
+
+    }
+
+    public static function bdAd_customizeDisplayAdCheck()
```

Comment:
Tên hàm này có vấn đề:

1. Tại sao cần prefix "bdAd_"? Nó là hàm trong `bdAd_Listener` rồi mà?
2. Tên hàm nên giải thích xem nó làm gì, ví dụ tên hàm "customize xyz" thì sẽ hiểu là gọi hàm này để biến đổi hiển thị cái gì đó. Nhưng hóa ra gọi hàm này để kiểm tra xem có hiện ads không -> không liên quan tí nào.
3. Thường với hàm kiểm tra flag boolean thì nên để prefix là "is", ví dụ hàm này có thể đặt là "isNoAd" hoặc "isAdFree". Tương tự, cái flag cũng sẽ đặt tên tương ứng với cái method "isNoAd" -> "$_isNoAd", "isAdFree" -> "$_isAdFree".